File: backend/app_tuyen_sinh_django/utils/enums/otp_type.py
from enum import Enum, auto


class OtpType(Enum):
    AUTH = auto()
    PASSWORD = auto()

    # def __init__(self) -> None:
    #     __class__.range_val = range(0, len(__class__))
    #     super().__init__()

    @property
    def otp_type(self):
        return self.name.capitalize()

    @staticmethod
    def from_id(input_id: int):
        try:
            return next(x for x in __class__ if x.value == int(input_id))
        except:
            return None

File: app_tuyen_sinh_flutter/lib/utils/enums/otp_type.dart
enum OtpType {
  auth(id: 1, otpType: 'Xác thực'),
  password(id: 2, otpType: 'Mật khẩu');

  final int id;
  final String otpType;

  const OtpType({
    required this.id,
    required this.otpType,
  });
}
